lib/gameteam.rb

```diff
@@ -141,5 +141,62 @@ def self.avg_scores_per_team_home
     end
     result_hash = team_scores.keys.zip(array_of_scores_to_games).to_h
   end
+
+  def self.pulls_team_id_max_score_home
+    avg_scores_per_team_home.max_by {|team_id, goals| goals}.first
+  end
+  
+  def self.pulls_team_id_min_score_home
+    avg_scores_per_team_home.min_by {|team_id, goals| goals}.first
+  end
+
+  def self.pull_id_goals_shots_and_math
+    id_and_goals = Hash.new(0)
+    id_and_shots = Hash.new(0)
+    
+    @@all.each do |row|
+      id_and_goals[row.team_id] += row.goals
+      id_and_shots[row.team_id] += row.shots
+    end
+
+    array_of_shots_to_goals =[]
+    result_hash = {}
+    id_and_shots.values.each_with_index do |value, index|
+      array_of_shots_to_goals << (value.to_f / id_and_goals.values[index]).round(2)
+    end
+    result_hash = id_and_shots.keys.zip(array_of_shots_to_goals).to_h
+  end
+
+  def self.most_accurate_team
```

Comment:
Most and least accurate team are supposed to accept an argument of season ID and then return the team with the best/worst accuracy.  These fail the Spec Harness for that reason too.